### PR TITLE
Auto convert speaker social profiles usernames to links

### PR DESCRIPTION
**Pull request by aviaryan**
Issue #363

Currently there is ambiguity whether to add username in speaker's social fields (github, twitter, facebook) or full profile link.
I saw both were used at FOSSASIA 2016 speaker's list (on [Open event server](http://open-event.herokuapp.com/admin/event/15/speaker) and in the google spreadsheet).
This required extra steps at the app's side (web and android) to handle those fields.

With this patch, I have added the feature that automatically convert usernames to links when speaker's data is updated or inserted.

### CODE REVIEW

**Comment by aviaryan** [comment]
@rafalkowalski @leto Please have a look at this patch. Thanks

**Comment by championswimmer** [comment]
So basically, 

Linkedin:
if only <string> -> make linkedin.com/in/<string>

G+ 
if only <string> -> plus.google.com/+<string>

fb:
facebook.com/<string>

twitter
twitter.com/<string> 

Also, if <string>[0] = "@" then drop that (can happen for Twitter) 

If want to have this functionality, let's implement all of it.

**Comment by championswimmer** [comment]
@mananwason @angularboy @SaptakS @the-dagger @shivenmian @opticod Please note, if this get merged, we need to make appropriate changes on the client side too. 
(Tagging all of you, so you'll be notified when this gets merged, and we can merge appropriate client side change too)

**Comment by SaptakS** [comment]
Yeah. A similar change would need to be merged in both the webapp and the android I guess.

**Comment by aviaryan** [comment]
> Linkedin:
if only -> make linkedin.com/in/

:+1:

> G+
if only -> plus.google.com/+

Currently google plus has both usernames as well as numeric ID's . Numeric ID's are for those who haven't created their public username on g+. The link is of the format https://plus.google.com/u/0/NUMBER (eg mine https://plus.google.com/u/0/110328513842183229282) . Note that NUMBER in my url changes to +USERNAME when you open it. We can handle both the scenarios (ID and username) but this will require some conditioning statements and the solution may still be ambiguous. What do you suggest ?

> Also, if [0] = "@" then drop that (can happen for Twitter) 

There's no need for that because Twitter automatically handles twitter.com/@username type urls. I guess the Twitter android app handles that too.

**Comment by championswimmer** [comment]
Regarding G+ numbers - 
They way understand the UX of this, either you 'remember' your handle, in which case you type from **memory**, and that only is valid for +UserName types. 

The number has ofc to be copied from an URL, in which case I presume the user is in a position to copy his whole profile url http://...

**Comment by aviaryan** [comment]
@championswimmer You are right. I will update this PR with the enhancements discussed above. 

PS - Don't know when this PR got closed. A bug in the android app maybe.

**Comment by aviaryan** [comment]
@championswimmer Updated the PR with Linkedin's url fix. There isn't a google plus field in speaker ( :open_mouth: ) so that wasn't needed.

File: open_event/models/speaker.py [hunk 1]
```diff
@@ -71,3 +73,19 @@ def serialize(self):
...
                 'sessions': session_data  }
+
+    def ensure_socialLinks(self):
+        """convert usernames in social network fields to full links"""
+        self.twitter = self._ensure_socialLink('https://twitter.com', self.twitter)
+        self.facebook = self._ensure_socialLink('https://www.facebook.com', self.facebook)
+        self.github = self._ensure_socialLink('https://github.com', self.github)
+        self.linkedin = self._ensure_socialLink('https://www.linkedin.com/in', self.linkedin)
+
+    def _ensure_socialLink(self, website, link):
```
Comment (juslee) [inline open_event/models/speaker.py]: Codacy Issue found: [Method could be a function](https://www.codacy.com/app/triplez/open-event-orga-server/file/2406529482/issues/source?bid=3233864&fileBranchId=3233886#l84)

**Comment by aviaryan** [comment]
@juslee Fixed the issue pointed by Codacy

File: open_event/models/speaker.py [hunk 2]
```diff
@@ -74,3 +77,10 @@ def serialize(self):
                 'position': self.position,
                 'country': self.country,
                 'sessions': session_data  }
+
```
Comment (juslee) [inline open_event/models/speaker.py]: Codacy Issue found: [Invalid method name "ensure_socialLinks"](https://www.codacy.com/app/triplez/open-event-orga-server/file/2411932579/issues/source?bid=3233864&fileBranchId=3233886#l81)

File: open_event/helpers/helpers.py [hunk 3]
```diff
@@ -67,3 +67,15 @@ def is_event_admin(event_id, users):
                         return is_admin
     return is_admin
 
+
+def ensure_socialLink(website, link):
```
Comment (juslee) [inline open_event/helpers/helpers.py]: Codacy Issue found: [Invalid function name "ensure_socialLink"](https://www.codacy.com/app/triplez/open-event-orga-server/file/2411932568/issues/source?bid=3233864&fileBranchId=3233886#l71)

**Comment by aviaryan** [comment]
@juslee please review. Issue #363

**Comment by rafalkowalski** [comment]
@aviaryan @juslee we need this i think in branch gsoc2016

**Comment by aviaryan** [comment]
@rafalkowalski Why :question:  :confused:

**Comment by SaptakS** [comment]
Seems good to merge.

**Comment by mariobehling** [comment]
I am confused why this gets merged to master, when the flow is: A) merge to Development Branch and B) Later to Master.

**Pull request merged**